app/grandchallenge/reader_studies/views.py

```diff
@@ -236,13 +240,67 @@ class ReadersUpdate(ReaderStudyUserGroupUpdateMixin):
     success_message = "Readers successfully updated"
 
 
-class ReaderStudyViewSet(ReadOnlyModelViewSet):
+class ExportCSVMixin(object):
+    def _check_export_perms(self, user, obj):
+        if not (user and user.has_perm(self.export_permission, obj)):
+            raise Http404()
+
+    def _preprocess_data(self, data):
+        processed = []
+        for entry in data:
+            processed.append(
+                map(lambda x: re.sub(r"[\n\r\t]", " ", str(x)), entry)
+            )
+        return processed
+
+    def _create_csv_response(self, data, filename="export.csv"):
+        response = HttpResponse(content_type="text/csv")
+        response["Content-Disposition"] = f'attachment; filename="{filename}"'
+        writer = csv.writer(response, quoting=csv.QUOTE_ALL, escapechar="\\")
+        writer.writerows(self._preprocess_data(data))
+
+        return response
+
+
+class ReaderStudyViewSet(ExportCSVMixin, ReadOnlyModelViewSet):
     serializer_class = ReaderStudySerializer
     queryset = ReaderStudy.objects.all().prefetch_related(
         "images", "questions"
     )
     permission_classes = [DjangoObjectPermissions]
     filter_backends = [ObjectPermissionsFilter]
+    export_permission = (
+        f"{ReaderStudy._meta.app_label}.change_{ReaderStudy._meta.model_name}"
+    )
+
+    @action(detail=True)
+    def export_questions(self, request, pk=None):
+        reader_study = self.get_object()
+        self._check_export_perms(request.user, reader_study)
+
+        data = []
+        for question in reader_study.questions.all():
+            data.append(
+                [question.question_text, question.get_answer_type_display()]
+            )
+
+        return self._create_csv_response(
+            data, filename=f"{reader_study.slug}-questions.csv"
+        )
+
+    @action(detail=True)
+    def export_answers(self, request, pk=None):
+        reader_study = self.get_object()
+        self._check_export_perms(request.user, reader_study)
+
+        fields = ["creator__username", "question__question_text", "answer"]
+        data = Answer.objects.filter(
+            question__reader_study=reader_study
+        ).values_list(*fields)
```

Comment:
The `Images` field is missing here - this is required to know what images this answer is for. The best way to serialize this for the user here is to use the images `name` as this should be unique for the reader study.

Reply:
Added and moved all fields to the answers export